### ISSUE TITLE
Dock not showing when set to Always Hide

### ISSUE THREAD

**Issue by joeljosedev** [pop-os/cosmic-dock#103]
![Screenshot from 2021-12-19 23-45-06](https://user-images.githubusercontent.com/53969297/146687292-6d771b24-0e03-4d73-b683-4173562a28d5.png)

**(1) Issue/Bug Description**

The dock does not pop up when the cursor is moved to the location of the dock in Always Hide mode. 

**(2) Steps to reproduce**
- Open Settings and set dock to Always Hide mode.
- Move cursor to the location of the dock

**(3) Expected behaviour**

The dock should pop up from its location

**(4) Distribution**
 
```
NAME="Pop!_OS"
VERSION="21.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 21.10"
VERSION_ID="21.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=impish
UBUNTU_CODENAME=impish
LOGO=distributor-logo-pop-os
```

**(5) Gnome Shell version**

GNOME Shell 40.5

**(6) Pop Shell version**

```
pop-shell:
  Installed: 1.1.0~1639174177~21.10~de901ae
  Candidate: 1.1.0~1639174177~21.10~de901ae
  Version table:
 *** 1.1.0~1639174177~21.10~de901ae 1001
       1001 http://apt.pop-os.org/release impish/main amd64 Packages
       1001 http://apt.pop-os.org/release impish/main i386 Packages
        100 /var/lib/dpkg/status
```

**(7) Where was Pop Shell installed from:**

Official ISO image from the Pop!_OS website

**(8) Monitor Setup**

1x 1080p, Primary(Horizontal) @ 60 Hz

**(9) Other Installed/Enabled Extensions**

None

**Comment by joeljosedev** [comment]
I made a bootable USB with Pop!_OS 21.10, booted into it and set the dock to Always Hide in the Settings app. The dock does not appear in this mode either.

**Comment by jacobgkau** [comment]
> I am running Pop!_OS 21.10 as a VM on VMware Workstation running on a Windows 11 host though.

> I made a bootable USB with Pop!_OS 21.10, booted into it and set the dock to Always Hide in the Settings app.

Did you boot into it on your physical hardware, or in VMware? Virtual machine managers may not work with pressure-sensing hot corners/screen edges, depending on how they handle mouse pointer integration.

**Comment by joeljosedev** [comment]
I booted into Pop!_OS with a live USB and also on a VM running on a Windows 11 host. The dock did not appear in both cases.

**Comment by jacobgkau** [comment]
You are moving the mouse past the edge of the screen, not just stopping where the dock would appear, correct?

Does this issue not occur when you're using `Intelligently Hide` and the dock is obstructed by a window?

What type of mouse are you using?

**Comment by joeljosedev** [comment]
I'm moving the cursor past the edge of the screen and it still doesn't work. When in `Intelligently Hide` mode, the dock hides when an app window covers it. I'm using the trackpad built into my ASUS laptop (ASUS Vivobook 15 X512FA).

**Comment by jacobgkau** [comment]
> When in `Intelligently Hide` mode, the dock hides when an app window covers it.

That is expected, but you should be able to move the mouse cursor to the edge and show it anyway. Does that also not work?

**Comment by joeljosedev** [comment]
`Intelligently Hide` mode works as expected. The problem is that the dock never pops up when set to `Always Hide` mode. I have tested this on a live environment as well as in a VM and have the same problem in both cases. I did try moving the mouse past the bottom of the screen but the dock remains hidden.

**Comment by joeljosedev** [comment]
I booted into a live USB with Pop!_OS 21.10 again, and the `Always Hide` mode worked this time. I'm not sure why it didn't work the last time I tried it, but it is working now, so I'll close this issue.

**Comment by leviport** [comment]
reopening for https://github.com/pop-os/linux/issues/114

**Comment by joeljosedev** [comment]
The problem I had was caused by VMware Workstation. The cursor didn't move beyond the screen window and that's why the dock never popped back up. When the cursor was moved outside the window, the control was transferred to the host OS and the cursor in the VM remained at the edge of the screen. I used the same ISO image, created a live USB, booted into it and had no problems with the dock.

**Comment by aindriu80** [comment]
I am having the same problem but on an installed version of Pop OS 21.10.  I have tried Dash to Dock and Cosmic Dock but I still have the same problem - intelligently hiding the dock results in the dock not appearing when I have the mouse placed over it, I have tried trackpad and USB mouse.

**Comment by Pentaphon** [comment]
My issue is that when I use "always hide" the dock doesn't appear every single time I drag my mouse to the bottom of the screen.

Not sure if I should make a new issue for that but thought I should mention that the dock isn't 100% reliable when moused over.

**Comment by ivanmonteiro** [comment]
I have the same problem - using PopOS 22.04 virtualized with Oracle Virtualbox on a Windows 10 host

**Comment by aindriu80** [comment]
I'm still having this problem, I'm using Pop 22.04 on an XPS 9550 laptop with 'Always Hide' enabled, the Cosmic Dock is not showing up for the majority of sessions.  I need to disable and enable extensions to get the dock to display again.  

I've tried many things (currently disabling Pop Shell) but no luck.  How can I determine the cause of this bug?  It's very frustrating.

![2022-06-08_13-50](https://user-images.githubusercontent.com/8885837/172621190-a6f52371-e34e-499a-895c-7d9cebd92556.png)


![2022-06-08_13-52](https://user-images.githubusercontent.com/8885837/172621213-d8c2c36d-09fb-4c9a-94cc-0c9aec15119c.png)

**Comment by ivanmonteiro** [comment]
Any workarounds?

**Comment by aindriu80** [comment]
I haven't found any, I wonder if reinstalling POP from scratch would fix it but I have a dual boot and have important files

**edit, yes reinstalling POP fixed it for me

**Comment by bassmanitram** [comment]
Starred - but I have to say it wasn't happening immediately I upgraded to 22.4 a few weeks ago - it's only started occurring in the last few days - a dodgy update?

_HOWEVER_ I DID just get "Always hide" working by doing this:

* Disable the dock in system sittings
* ALT-F2, r to restart gnome shell
* Enable the dock (didn't make it work)
* Change from "All Displays" to "Primary Display" - now it works

Not sure if it made a difference but during this procedure, I made sure that NONE of my displays were showing a full-screen window.

Now it appears to work on the Primary Display, fullscreen window or not

**Comment by akikanellis** [comment]
For anyone having this issue on VirtualBox, a workaround you can use is to disable the mouse integration feature of VirtualBox from the _Input -> Mouse Integration_ setting.

This will allow your cursor to be fully captured by the VM which will make the dock behave as expected on the "Always visible" and "Intelligently hide" options. The drawback of this is that you need to "release" the cursor with the _Host key_ (usually _Right Ctrl_) whenever you want to use the cursor on the host machine.

**Comment by ouroboroscoding** [comment]
I am having the exact same issue on 22.04, but using Synergy, a tool to share your mouse/keyboard across computers. No matter what I do, adding a huge dead spot across half the screen, adding a timeout, etc, the dock will just not show up on the secondary computer. It's as if it knows the difference between a mouse position set by a physical mouse, and one set by software, and for some reason is ignoring the latter.

**Comment by Jeanball** [comment]
Same issue on my side. 

Someone found a solution? 

I'm using Synergy (Tried Barrier before) and cannot recognize my mouse when I want to show up the dock.

Right now, I disable Cosmic dock to use dock to panel but really want to use cosmic dock one day with my KVM software.

**Comment by geobam** [comment]
I had the same issue with the Option "always hide". But when i move the cursor with faster speed to the bottom edge the dock appears. But not when i move the cursor slowly. Is there maybe a config to make it more sensible?

Edit: 
Found a solution for me in that thread here: https://github.com/pop-os/cosmic-dock/issues/162

TL;DR:
Either use dconf Editor and change /org/gnome/shell/extensions/dash-to-dock/pressure-threshold to a value you feel comfortable (for me it was 10) or use the CLI for that `gsettings set org.gnome.shell.extensions.dash-to-dock pressure-threshold 10`

**Comment by penguinlab** [comment]
The same problem occurs in my environment.
I am using a virtual machine created with Proxmox VE, connected via Remote Desktop.
It is very inconvenient.
